src/CBF/CBF.jl

```diff
@@ -0,0 +1,588 @@
+            println(io, "BCOORD")
+            println(io, length(bcoord))
+            for (a, v) in bcoord
+                println(io, a-1, " ", v)
+            end
+            println(io)
+        end
+
+        # PSD constraints.
+        psdcon = Int[] # List of side dimensions.
+        hcoord = Tuple{Int, Int, Int, Int, Float64}[] # Affine terms.
+        dcoord = Tuple{Int, Int, Int, Float64}[] # Constant terms.
+
+        for ci in getmodelcons(MOI.VectorOfVariables,
+            MOI.PositiveSemidefiniteConeTriangle) # MOI variable constraints.
+            side = getconset(ci).side_dimension
+            push!(psdcon, side)
+            vars = getconfun(ci).variables
+            k = 0
+            for i in 1:side, j in 1:i
+                k += 1
+                push!(hcoord, (length(psdcon), vars[k].value, i, j, 1.0))
+            end
+            @assert k == length(vars)
+        end
+
+        for ci in getmodelcons(MOI.VectorAffineFunction{Float64},
+            MOI.PositiveSemidefiniteConeTriangle) # MOI affine constraints.
+            side = getconset(ci).side_dimension
+            push!(psdcon, side)
+            fi = getconfun(ci)
+            for vt in fi.terms
+                st = vt.scalar_term
+                # Get (i,j) index in symmetric matrix from lower triangle index.
+                i = div(1 + isqrt(8*vt.output_index - 7), 2)
+                j = vt.output_index - div(i*(i-1), 2)
+                push!(hcoord, (length(psdcon), st.variable_index.value, i, j,
+                    st.coefficient))
+            end
+            k = 0
+            for i in 1:side, j in 1:i
+                k += 1
+                if !iszero(fi.constants[k])
+                    push!(dcoord, (length(psdcon), i, j, fi.constants[k]))
+                end
+            end
+            @assert k == MOI.output_dimension(fi)
+        end
+
+        if !isempty(psdcon)
+            println(io, "PSDCON")
+            println(io, length(psdcon))
+            for v in psdcon
+                println(io, v)
+            end
+            println(io)
+        end
+
+        if !isempty(hcoord)
+            println(io, "HCOORD")
+            println(io, length(hcoord))
+            for (a, b, c, d, v) in hcoord
+                println(io, a-1, " ", b-1, " ", c-1, " ", d-1, " ", v)
+            end
+            println(io)
+        end
+
+        if !isempty(dcoord)
+            println(io, "DCOORD")
+            println(io, length(dcoord))
+            for (a, b, c, v) in dcoord
+                println(io, a-1, " ", b-1, " ", c-1, " ", v)
+            end
+            println(io)
+        end
+    end
+    
+    return
+end
+
+# ==============================================================================
+#
+#   MOI.read_from_file
+#
+# The CBF file format (version 3) is described at
+# http://cblib.zib.de/doc/format3.pdf.
+#
+# ==============================================================================
+
+function mattovecidx(i, j)
```

Comment:
Docstrings or comments on non-obvious functions. Also, can we name this something more obvious?
`matrix_to_vector_index`?